Replace “xkill” with “qvm-xkill”

DemiMarie opened the pull request:
`qvm-xkill` can handle override redirect windows, which `xkill` cannot.

Fixes https://github.com/QubesOS/qubes-issues/issues/6235

DemiMarie committed:
Replace “xkill” with “qvm-xkill”

“qvm-xkill” can handle override redirect windows, which “xkill” cannot.

marmarek reviewed (changes requested):
This fixes default configuration, but sadly XFCE copies the config to the user home at the first start and then ignores further updates of the system one. This means, the fix won't work for existing users (both current R4.1 users, and later those updating R4.0 -> R4.1). And will be useless if we decide to backport it to R4.0.

I don't know any nice method of handling such defaults changes, but we did that in the past - see #12. You can re-use the same script, it is supposed to handle subsequent updates by storing patch version in the flag file (currently at "1").

DemiMarie committed:
Update existing XFCE configurations

XFCE copies the default configuration into the user’s home directory,
and afterwards ignores changes to the defaults.  Fortunately,
‘update-xfce-config’ already takes care of updating existing
configurations.  Extend it to replace the default “xkill” keybinding
with “qvm-xkill”.

While I was working on this, I added locking to prevent concurrent
executions, and a Vim modeline so that the formatting is preserved.

marmarek commented on update-xfce-config:
```diff
@@ -28,6 +36,16 @@ if [ "$UPDATE_CONFIG_DONE" -lt 1 ]; then
+if [ "$UPDATE_CONFIG_DONE" -lt 2 ]; then
+    reset_xfconf_if_default \
+        "xfce4-keyboard-shortcuts" \
+        "/commands/custom/<Control><Alt>Esc" \
```
This fails for me with:
```
Property "/commands/custom/<Control><Alt>Esc" does not exist on cannel "xfce4-keyboard-shortcuts".
```
Looking at `xfconf-query -c xfce4-keyboard-shortcuts` output, I do have `/commands/default/<Control><Alt>Escape` (note `custom` -> `default` and `Esc` -> `Escape`). But then, if I manually add `qvm-xkill` command, I get extra property `/commands/custom/<Control><Alt>Escape` (but the one with "default" remains, with value `xkill`).
I'm not really sure what is the proper way to handle this (whether to modify default or to add custom).

DemiMarie replied:
I believe we should modify the default, so that it will be reflected if the user ever reverts to the default configuration.

marmarek commented:
This is better now, but still not ideal. Now only the default value changes, which means I need to go to the settings then reset shortcuts to default for this to take effect.

DemiMarie committed:
Update both default and custom configurations

This ensures that if users revert to defaults, they won’t get old
settings.

DemiMarie commented:
@marmarek Fixed by updating both default and custom configurations, for all of the shortcuts.
